### PR TITLE
Improve JS formatting and HTML content

### PR DESCRIPTION
**Pull request by chrisdavidmills**
A few things were annoying me about my recently-added JS builtin examples, which I've fixed in this PR:

- The HTML content was left as my default template content "My template", which looked a bit unfinished/unprofessional. I've replaced those with a short description of what each example is doing.
- The spacing in the JS was 4 spaces, and we tend to use 2 in MDN code.
- I've updated the comments in the JavaScript files to be consistent with those used in the MDN page JavaScript snippets.

### CODE REVIEW

File: js-builtin-examples/compile-streaming/index.js
Comment (bsmth) [inline js-builtin-examples/compile-streaming/index.js:1]: Not sure if you want to change it, but Prettier adds trailing commas:

```diff
-    log: console.log
-  }
+    log: console.log,
+  },
```

And wraps in parens for `arrowParens` option: -> https://prettier.io/docs/en/options.html#arrow-function-parentheses

```diff
-  .then(response => response.arrayBuffer())
-  .then(bytes => WebAssembly.compile(bytes, compileOptions))
-  .then(module => WebAssembly.instantiate(module, importObject))
-  .then(instance => instance.exports.main());
+  .then((response) => response.arrayBuffer())
+  .then((bytes) => WebAssembly.compile(bytes, compileOptions))
+  .then((module) => WebAssembly.instantiate(module, importObject))
+  .then((instance) => instance.exports.main());
```

File: js-builtin-examples/compile-streaming/index.html [hunk 1]
```diff
@@ -9,7 +9,8 @@
 </head>
 
 <body>
-  <p>My template</p>
+  <p>Exported Wasm function concatenates two strings (via a JS String builtin) and logs the result to the JavaScript
```
Comment (bsmth) [inline js-builtin-examples/compile-streaming/index.html:12]: Missed this first time around, thanks for adding 👍🏻

File: js-builtin-examples/compile-streaming/index.js
Comment (chrisdavidmills) [reply js-builtin-examples/compile-streaming/index.js:1]: Yeah, good to be consistent; done!

We also decided to change the imported string constant namespace in the docs from `"#"` to `"string_constants"` so it is more readable/intuitive, so I've updated it in all the example files too.

**Review by bsmth — APPROVED** [review]
Thanks!

**Pull request merged**